tests/ui/erasing_op.rs

```diff
@@ -1,3 +1,13 @@
+struct Length(u8);
+struct Meter;
+
+impl core::ops::Mul<Meter> for u8 {
+    type Output = Length;
+    fn mul(self, _: Meter) -> Length {
+        Length(self)
+    }
+}
```

Comment:
The example in the issue was used to create a new type, like the one you added here. Could you add a second test case there the argument stays the same, to ensure that operations returning the same type is still being linted? You can use this:

```rs
#![allow(unused)]

struct Vec2 {
    x: u32,
    y: u32,
}

impl core::ops::Mul<Vec2> for u8 {
    type Output = Vec2;
    fn mul(self, _: Vec2) -> Vec2 {
        Vec2 {
            x: 0,
            y: 0,
        }
    }
}
```

Reply:
Thanks for the prompt feedback even in these holiday seasons. I have added something similar, but the current logic allows to scale such a vector by a constant 0. If you have a good idea how to make the 2 extra test cases I have added fail the lint without failing `0 * Meter`, feel free to advise or just commit on my branch.

There are 3 types involved in a binary op, and it is easy to create false positives or ignore something relevant. If one of the arguments is a constant integer 0, I would test the other argument's type against the output type, and ignore the lint if those are different. But I am unsure whether that would be a sufficient implementation.

Reply:
You're welcome! I'm currently using Clippy to take a break from some other work that I should do, it's a little like a holiday. ^^

Regarding your suggestion, that would actually be the route I'd take. You're right that it might have some disadvantages, even if I can't really think of any ATM. The current suggestion isn't concerned with the return type and lints in either case, looking at the open issues it seems like the issue you're fixing was the only relevant report. Therefore, I think it's safe to say that doing this comparison would be a good fix as it increases the precision of the lint. So, if you have the time, it would be perfect if you could implement your suggestion :upside_down_face: